src/lambda/wizards/samInitWizard.ts

```diff
@@ -0,0 +1,209 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as os from 'os'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { SamCliInitArgs } from '../../shared/sam/cli/samCliInit'
+import * as lambdaRuntime from '../models/lambdaRuntime'
+import { MultiStepWizard, WizardStep } from '../wizards/multiStepWizard'
+
+export interface CreateNewSamAppWizardContext {
+    readonly lambdaRuntimes: lambdaRuntime.LambdaRuntime[]
+    readonly workspaceFolders: vscode.WorkspaceFolder[] | undefined
+
+    showInputBox(
+        options?: vscode.InputBoxOptions,
+        token?: vscode.CancellationToken
+    ): Thenable<string | undefined>
+
+    showOpenDialog(
+        options: vscode.OpenDialogOptions
+    ): Thenable<vscode.Uri[] | undefined>
+
+    showQuickPick(
+        items: string[] | Thenable<string[]>,
+        options: vscode.QuickPickOptions & { canPickMany: true },
+        token?: vscode.CancellationToken
+    ): Thenable<string[] | undefined>
+    showQuickPick(
+        items: string[] | Thenable<string[]>,
+        options?: vscode.QuickPickOptions,
+        token?: vscode.CancellationToken
+    ): Thenable<string | undefined>
+    showQuickPick<T extends vscode.QuickPickItem>(
+        items: T[] | Thenable<T[]>,
+        options: vscode.QuickPickOptions & { canPickMany: true },
+        token?: vscode.CancellationToken
+    ): Thenable<T[] | undefined>
+    showQuickPick<T extends vscode.QuickPickItem>(
+        items: T[] | Thenable<T[]>,
+        options?: vscode.QuickPickOptions,
+        token?: vscode.CancellationToken
+    ): Thenable<T | undefined>
+}
+
+class DefaultCreateNewSamAppWizardContext implements CreateNewSamAppWizardContext {
+    public readonly lambdaRuntimes = lambdaRuntime.lambdaRuntimes
+    public readonly showInputBox = vscode.window.showInputBox
+    public readonly showOpenDialog = vscode.window.showOpenDialog
+    public readonly showQuickPick = vscode.window.showQuickPick
+
+    public get workspaceFolders(): vscode.WorkspaceFolder[] | undefined {
+        return vscode.workspace.workspaceFolders
+    }
+}
+
+export class CreateNewSamAppWizard extends MultiStepWizard<SamCliInitArgs> {
+    private readonly RUNTIME: WizardStep = new WizardStep(async () => {
+        this.runtime = await this.context.showQuickPick(this.context.lambdaRuntimes, {
```

Comment:
Please alphanumerically sort the items going into the selection UI if possible

Reply:
These items are sorted by page order in the Wizard. I think sorting by page order to more appropriate than sorting alphanumerically here.

Reply:
I'm not sure what you mean by page order. 
What I meant is when the wizard shows the runtimes to the user, the list of runtimes should be alphabetized

Reply:
Oh, you mean the choices themselves. Included in the latest commit.

Reply:
✔️